src/AutoRest.CSharp/LowLevel/Generation/LowLevelDataPlaneClientWriter.cs

```diff
@@ -0,0 +1,701 @@
+                }
+            }
+        }
+
+        private void WriteClientMethod(CodeWriter writer, LowLevelClientMethod clientMethod, bool async)
+        {
+            WriteClientMethodDecleration(writer, clientMethod.RestMethod, clientMethod.OperationSchemas, async);
+            WriteClientMethodBody(writer, clientMethod, async);
+        }
+
+        private void WritePagingMethod(CodeWriter writer, LowLevelPagingMethod clientMethod, bool async)
+        {
+            var pageMethodReturnType = async ? new CSharpType(typeof(Task<Page<BinaryData>>)) : new CSharpType(typeof(Page<BinaryData>));
+            var asyncText = async ? "async " : string.Empty;
+            var awaitText = async ? "await " : string.Empty;
+            var enumerableFactoryMethod = async ? "CreateAsyncEnumerable" : "CreateEnumerable";
+
+            WriteClientMethodDecleration(writer, clientMethod.FirstPageMethod, clientMethod.OperationSchemas, async);
+
+            using (writer.Scope())
+            {
+                writer.Line($"options ??= new {typeof(Azure.RequestOptions)}();");
+
+                WriteEnumerableArgumentEnumerationStatements(writer, clientMethod.FirstPageMethod, out var enumeratedParameterMap);
+
+                writer.Line($"{asyncText}{pageMethodReturnType} FirstPageFunc(int? pageSizeHint)");
+                using (writer.Scope())
+                {
+                    WritePagingFuncMethodBody(writer, clientMethod.Diagnostic, clientMethod.FirstPageMethod, clientMethod.PagingResponseInfo, async, enumeratedParameterMap);
+                }
+                writer.Line();
+
+                if (clientMethod.PagingResponseInfo.NextPageMethod != null)
+                {
+                    writer.Line($"{asyncText}{pageMethodReturnType} NextPageFunc(string nextLink, int? pageSizeHint)");
+                    using (writer.Scope())
+                    {
+                        WritePagingFuncMethodBody(writer, clientMethod.Diagnostic, clientMethod.PagingResponseInfo.NextPageMethod, clientMethod.PagingResponseInfo, async, enumeratedParameterMap);
+                    }
+                    writer.Line();
+                }
+
+                writer.Line($"return PageableHelpers.{enumerableFactoryMethod}(FirstPageFunc, {(clientMethod.PagingResponseInfo.NextPageMethod != null ? "NextPageFunc" : "null")});");
+            }
+
+            writer.Line();
+        }
+
+        private bool IsIEnumerableType(CSharpType type)
+        {
+            return type.IsFrameworkType && (type.FrameworkType == typeof(IEnumerable) || (type.FrameworkType.IsGenericType && type.FrameworkType.GetGenericTypeDefinition() == typeof(IEnumerable<>)));
+        }
+
+        private void WriteEnumerableArgumentEnumerationStatements(CodeWriter writer, RestClientMethod clientMethod, out Dictionary<string, CodeWriterDeclaration> enumeratedParameterMap)
+        {
+            enumeratedParameterMap = new Dictionary<string, CodeWriterDeclaration>();
```

Comment:
Why `out` instead of just return?

Reply:
I will change it.  My thought here was that it would be a little strange to have a non void returning `WriteXyz` method, but I'll just use a `return` here.

Reply:
Fixed.